crates/solver/src/liquidity/zeroex.rs

```diff
@@ -9,39 +9,287 @@ use {
         liquidity_collector::LiquidityCollecting,
         settlement::SettlementEncoder,
     },
-    anyhow::Result,
+    anyhow::{anyhow, Context, Result},
     contracts::{GPv2Settlement, IZeroEx},
+    ethcontract::Address,
+    futures::{SinkExt, StreamExt},
     model::{order::OrderKind, TokenPair},
     primitive_types::{H160, U256},
+    reqwest::{IntoUrl, Url},
     shared::{
         ethrpc::Web3,
         http_solver::model::TokenAmount,
         recent_block_cache::Block,
-        zeroex_api::{Order, OrderRecord, OrdersQuery, ZeroExApi},
+        zeroex_api::{
+            websocket::{OrderRecord, OrderState, OrdersResponse},
+            Order,
+            OrdersQuery,
+            ZeroExApi,
+        },
     },
     std::{
         collections::{HashMap, HashSet},
         sync::Arc,
+        time::Duration,
     },
+    tokio::{net::TcpStream, sync::Mutex},
```

Comment:
A `std::Mutex` is sufficient at the moment. Also if we are reusing that code in the `driver` (which might try to fetch liquidity in parallel due to multiple solvers) a `RwLock` could also make sense.